etcdserver/server.go

```diff
@@ -223,27 +223,36 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 	}
 
 	haveWAL := wal.Exist(cfg.WALDir())
+	if !haveWAL {
+		defer func() {
+			if err != nil {
+				// cleans up member directory if bootstrap fails (including forming or joining a new cluster)
+				os.RemoveAll(cfg.MemberDir())
+			}
+		}()
+	}
 	ss := snap.New(cfg.SnapDir())
 
-	prt, err := rafthttp.NewRoundTripper(cfg.PeerTLSInfo, cfg.peerDialTimeout())
-	if err != nil {
-		return nil, err
+	prt, uerr := rafthttp.NewRoundTripper(cfg.PeerTLSInfo, cfg.peerDialTimeout())
```

Comment:
we should not call everything uerr. When there is a naming conflict for error, we usually add one prefix. For newRoundTripper, probably just rterr.